velo_action/gcp.py

```
@@ -12,25 +13,12 @@
 
 
 class GCP:
-    def __init__(self, service_account_key):
-        google_service_account_key_json_str = None
-        try:
-            google_service_account_key_json_str = base64.b64decode(
-                service_account_key.encode("ascii")
-            ).decode("ascii")
-        except Exception as e:
-            logger.debug("INPUT_SERVICE_ACCOUNT_KEY was not base64 encoded")
-
-        if google_service_account_key_json_str:
-            service_account_info = json.loads(google_service_account_key_json_str)
+    def __init__(self, service_account_key=None):
+        self.scoped_credentials = None
+        if service_account_key:
+            self._auth_service_account(service_account_key)
         else:
-            service_account_info = json.loads(service_account_key)
-        credentials = service_account.Credentials.from_service_account_info(
-            service_account_info
-        )
-        self.scoped_credentials = credentials.with_scopes(
-            ["https://www.googleapis.com/auth/cloud-platform"]
-        )
+            logger.info("Using local credentials")
```

What if no service account key is specified and local credentials dont exist?

We should return a nice error for the Github workflow log...

---

Yes, we can. The 403 you get from the gcp SDK is not so bad though.

---

That is:

```google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
```

But we should add a note for `INPUT_SERVICE_ACCOUNT_KEY`

---

Added a better error message.

For missing credentials: 

> RuntimeError: No valid credentials to access Google cloud. Please either specify the INPUT_SERVICE_ACCOUNT_KEY environment or authenticate using 'gcloud auth login'.


For lack of permissions:

> RuntimeError: Missing permission to access secret 'velo_action_octopus_server' in project 'nube-velo-prod'. Elevate your permissions with: klipy power elevate --group nube.project.editor.nube-velo-prod